Tests/Unit/NetworkServer/ModuleConnectionHostTest.cs

```diff
@@ -273,10 +274,79 @@ public async Task OnDirectMethodCall_ClearCache_When_Correct_Should_Work()
             networkServerConfiguration.IoTEdgeTimeout = 5;
 
             await using var moduleClient = new ModuleConnectionHost(networkServerConfiguration, classCMessageSender.Object, this.loRaModuleClientFactory.Object, loRaDeviceRegistry.Object, loRaDeviceApiServiceBase, NullLogger<ModuleConnectionHost>.Instance, TestMeter.Instance);
-            await moduleClient.OnDirectMethodCalled(new Microsoft.Azure.Devices.Client.MethodRequest(Constants.CloudToDeviceClearCache), null);
+            await moduleClient.OnDirectMethodCalled(new MethodRequest(Constants.CloudToDeviceClearCache), null);
             loRaDeviceRegistry.VerifyAll();
         }
 
+        [Fact]
+        public async Task OnDirectMethodCall_DropConnection_Should_Work_As_Expected()
+        {
+            // arrange
+            var networkServerConfiguration = new NetworkServerConfiguration();
+            var classCMessageSender = new Mock<IClassCDeviceMessageSender>(MockBehavior.Strict);
+            var loRaDeviceRegistry = new Mock<ILoRaDeviceRegistry>(MockBehavior.Strict);
+            var devEui = new DevEui(0);
+            var mockedDevice = new Mock<LoRaDevice>(null, devEui, null);
+            _ = loRaDeviceRegistry.Setup(x => x.GetDeviceByDevEUIAsync(devEui)).ReturnsAsync(mockedDevice.Object);
+            var c2d = $"{{\"DevEui\": \"{ devEui }\", \"Fport\": 1, \"MessageId\": \"{ Guid.NewGuid() }\"}}";
+
+            // act
+            await using var moduleClient = new ModuleConnectionHost(networkServerConfiguration, classCMessageSender.Object, this.loRaModuleClientFactory.Object, loRaDeviceRegistry.Object, loRaDeviceApiServiceBase, NullLogger<ModuleConnectionHost>.Instance, TestMeter.Instance);
+            await moduleClient.OnDirectMethodCalled(new MethodRequest(Constants.CloudToDeviceCloseConnection, Encoding.UTF8.GetBytes(c2d)), null);
+
+            // assert
+            loRaDeviceRegistry.VerifyAll();
+            mockedDevice.Verify(x => x.CloseConnectionAsync(It.IsAny<CancellationToken>(), true), Times.Once);
+        }
+
+        public static TheoryData<byte[]> DropConnectionInvalidMessages =>
+            new TheoryData<byte[]>
+                {
+                    null,
```

Comment:
Did you mean to test with a _null reference_ or a _JSON null_, as in `Encoding.UTF8.GetBytes("null")` in the latter case?

Reply:
As discussed, changed with 0be1f5b3cda6ab3abbfb0bb1b65006befb7139a5